lib/string_reverse.rb

```diff
@@ -1,4 +1,17 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+
+  while my_string != nil
+    i = 0
+    j = (my_string.length - 1)
+    while i < j
+      temp = my_string[j]
+      my_string[j] = my_string[i]
+      my_string[i] = temp
+      j -= 1
+      i += 1
+    end
+    return my_string

```

Comment:
Technically, since you're reversing the string in place, you don't need to return anything.